### ISSUE TITLE
Non-deterministic scatter reduction algorithms for scatter operations for CUDA (sum, subtract, divide, multiply). 

### ISSUE THREAD

**Issue by v0dro** [pytorch/pytorch#33394]
## 🚀 Feature

CUDA equivalent of https://github.com/pytorch/pytorch/issues/33389

New issue for the sake of clarity and brevity.

**Comment by rgommers** [comment]
@v0dro all these separate issues remains confusing, better not to do that. Right now this issue isn't linked at all to gh-22378. i believe once your PR gh-41977 is merged, the work for that issue is complete, right?
